src/common/browser-adapters/webextension-browser-adapter.ts

```diff
@@ -9,6 +18,20 @@ import { StorageAdapter } from './storage-adapter';
 export abstract class WebExtensionBrowserAdapter
     implements BrowserAdapter, StorageAdapter, CommandsAdapter
 {
+    private browserEventManager: BrowserEventManager;
+
+    constructor(
+        browserEventManager: BrowserEventManager,
+        browserEvents: DictionaryStringTo<Events.Event<any>>,
+    ) {
+        this.browserEventManager = browserEventManager;
+        for (const [eventType, event] of Object.entries(browserEvents)) {
+            if (event !== undefined) {
+                this.browserEventManager.addBrowserListener(event, eventType);
```

Comment:
It would be good leave a comment explaining that it's important that these calls happen synchronously from a service worker initializer; that probably won't be obvious to a future maintainer